landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,683 @@
+            # if nothing is passed with qsinlet_ts, qsinlet remains the same from
+            # initialized parameters
+            else:  # qsinlet_ts==None:
+                qsinlet = self.qsinlet
+                qs_in[inlet_node] = qsinlet
+            if inlet_area_ts is not None:
+                inlet_area = inlet_area_ts
+                runoffinlet = np.ones(grid.number_of_nodes) * grid.dx ** 2
+                # Change the runoff at the inlet node to node area + inlet node
+                runoffinlet[inlet_node] += inlet_area
+                _ = grid.add_field(
+                    "node", "water__unit_flux_in", runoffinlet, noclobber=False
+                )
+                # if inlet area has changed with time (so we have a new inlet area here)
+                fa = FlowAccumulator(
+                    grid,
+                    surface="topographic__elevation",
+                    flow_director="FlowDirectorD8",
+                    runoff_rate=None,
+                    depression_finder="DepressionFinderAndRouter",
+                    router="D8",
+                )
+                (da, q) = fa.accumulate_flow()
+                q = grid.at_node["surface_water__discharge"]
+                # this is the drainage area that I need for code below with an inlet set
+                # by spatially varible runoff.
+                da = q / grid.dx ** 2
+            else:  # inletarea not changing with time
+                q = grid.at_node["surface_water__discharge"]
+                # this is the drainage area that I need for code below with an inlet set
+                # by spatially varible runoff.
+                da = q / grid.dx ** 2
+        #                print ("da", da[inlet_node])
+        # if inlet flag is not on, proceed as normal.
+        else:
+            # renamed this drainage area set by flow router
+            da = grid.at_node["drainage_area"]
+        # flow__upstream_node_order is node array contianing downstream to
+        # upstream order list of node ids
+        s = grid.at_node["flow__upstream_node_order"]
+        max_slopes = grid.at_node["topographic__steepest_slope"]
+        flowdirs = grid.at_node["flow__receiver_node"]
+
+        # make a list l, where node status is interior (signified by label 0) in s
+        interior_s = s[np.where((grid.status_at_node[s] == 0))[0]]
+        dwnst_nodes = interior_s.copy()
+        # reverse list so we go from upstream to down stream
+        dwnst_nodes = dwnst_nodes[::-1]
+        max_slopes[:] = max_slopes.clip(0)
+        for i in dwnst_nodes:
+            # calc deposition and erosion
+            dep = alph * qs_in[i] / da[i]
+            ero = -Kv[i] * da[i] ** (0.5) * max_slopes[i]
+            dzver[i] = dep + ero
+            # potential lateral erosion initially set to 0
+            petlat = 0.0
+            # water depth in meters, needed for lateral erosion calc
+            wd = 0.4 * (da[i] * runoffms) ** 0.35
```

Comment:
Suggest either making the 0.4 and 0.35 as user-definable input parameters, or using syntactic sugar (define them as constants near the top of the file, then just refer to those constants here, e.g., define WIDTH_EXPONENT = 0.35 near the head of the file)

Reply:
Defined wid_coeff and wid_exp as constants near the head of the file.